src/debug/HyprCtl.cpp

```diff
@@ -945,7 +947,22 @@ std::string systemInfoRequest(eHyprCtlOutputFormat format, std::string request)
 #if defined(__DragonFly__) || defined(__FreeBSD__)
     const std::string GPUINFO = execAndGet("pciconf -lv | fgrep -A4 vga");
 #elif defined(__arm__) || defined(__aarch64__)
-    const std::string GPUINFO = execAndGet("cat /proc/device-tree/soc*/gpu*/compatible");
+    std::string                 GPUINFO;
+    const std::filesystem::path dev_tree = "/proc/device-tree";
+    if (std::filesystem::exists(dev_tree) && std::filesystem::is_directory(dev_tree)) {
```

Comment:
all of this might throw so probably should be in a try catch

Reply:
The entire `if` block? Just catch `...` (and I guess like `execAndGet` set `GPUINFO` to `"error"`) okay?

Reply:
every std::filesystem call can throw unless supplied with a std::error_code as the second arg, so the entire thing

Reply:
Alright, will do

Reply:
Done